src/cbmc_starter_kit/update.py

```diff
@@ -0,0 +1,211 @@
+        raise UserWarning(f'CBMC root is missing a {util.PROOF_DIR} subdirectory: {args.cbmc_root}')
+    logging.debug('CBMC root: %s', args.cbmc_root)
+    return args
+
+def validate_starter_kit_root(args):
+    starter_kit = repository.starter_kit_root(repo=args.cbmc_root, abspath=True)
+    args.starter_kit_root = args.starter_kit_root or starter_kit
+    if args.starter_kit_root:
+        args.starter_kit_root = args.starter_kit_root.resolve()
+        if not args.starter_kit_root.is_dir():
+            raise UserWarning(f'Starter kit root is not a directory: {args.starter_kit_root}')
+        if not (args.starter_kit_root/util.REPOSITORY_TEMPLATES).is_dir():
+            raise UserWarning(f'Starter kit root is missing a {util.REPOSITORY_TEMPLATES} '
+                              f'subdirectory: {args.starter_kit_root}')
+        if not args.starter_kit_root.is_relative_to(args.cbmc_root):
+            raise UserWarning(f'Starter kit root is {args.starter_kit_root} is not a descendant of '
+                              f'CBMC root {args.cbmc_root}')
+    else:
+        args.no_migrate = True
+    logging.debug('CBMC starter kit root: %s', args.starter_kit_root)
+    return args
+
+################################################################
+
+def files_under_root(root=None, symlinks_only=False):
+    if root and not root.is_dir():
+        logging.critical('Not a directory: %s', root)
+        sys.exit(1)
+
+    cmd = ['find', '.']
+    if symlinks_only:
+        cmd += ['-type', 'l']
+    kwds = {
+        'cwd': root,
+        'text': True,
+        'capture_output': True,
+    }
+    result = subprocess.run(cmd, **kwds, check=True)
+
+    if not result.returncode:
+        return sorted(result.stdout.splitlines())
+    logging.critical('This should be impossible: Failed to list files under directory: %s', root)
+    sys.exit(1)
+
+################################################################
+
+def migrate(cbmc_root, starter_kit_root):
+    logging.debug('Migrating CBMC starter kit')
+    templates = files_under_root(starter_kit_root/util.REPOSITORY_TEMPLATES)
+    logging.debug('Migrating CBMC starter kit: found templates: %s', templates)
+    symlinks = files_under_root(cbmc_root, symlinks_only=True)
+    logging.debug('Migrating CBMC starter kit: found symlinks: %s', symlinks)
+
+    cbmc_symlinks = [path for path in templates if path in symlinks]
+    for symlink in cbmc_symlinks:
+        cbmc_link = cbmc_root / symlink
+        cbmc_file = cbmc_link.resolve()
+        logging.warning('Copying: %s -> %s', cbmc_file, cbmc_link)
+        assert cbmc_file.exists()
+        assert cbmc_link.exists()
+        cbmc_link.unlink()
+        shutil.copy(cbmc_file, cbmc_link)
+
+def remove_negative_tests(cbmc_root):
+    logging.debug('Removing CBMC starter kit negative tests')
+    negative_tests = cbmc_root / util.NEGATIVE_TESTS
+    if negative_tests.is_dir():
+        logging.warning('Removing: %s', negative_tests)
```

Comment:
This also looks more like an "info" message to me.

Reply:
Changed to info.